### ISSUE TITLE
Problem migrating to 3.1

### ISSUE THREAD

**Issue by sliwowitz** [divio/django-cms#4119]
I have a problem running a migration from 3.0.(idontknow) to 3.1.0 After hours of fiddling with MIGRATION_MODULES, according to what I found elsewhere, I have:

```
MIGRATION_MODULES = {
    'easy_thumbnails': 'easy_thumbnails.migrations',
    'cms': 'cms.migrations_django',
    'menus': 'menus.migrations_django',
    'filer': 'filer.migrations_django',
    'djangocms_column': 'djangocms_column.migrations_django',
    'djangocms_file': 'djangocms_file.migrations_django',
    'djangocms_flash': 'djangocms_flash.migrations_django',
    'djangocms_googlemap': 'djangocms_googlemap.migrations_django',
    'djangocms_inherit': 'djangocms_inherit.migrations_django',
    'djangocms_link': 'djangocms_link.migrations_django',
    'djangocms_picture': 'djangocms_picture.migrations_django',
    'djangocms_snippet': 'djangocms_snippet.migrations_django',
    'djangocms_style': 'djangocms_style.migrations_django',
    'djangocms_teaser': 'djangocms_teaser.migrations_django',
    'djangocms_video': 'djangocms_video.migrations_django',
    'djangocms_text_ckeditor': 'djangocms_text_ckeditor.migrations_django',
    'cmsplugin_filer_file': 'cmsplugin_filer_file.migrations_django',
    'cmsplugin_filer_folder': 'cmsplugin_filer_folder.migrations_django',
    'cmsplugin_filer_link': 'cmsplugin_filer_link.migrations_django',
    'cmsplugin_filer_image': 'cmsplugin_filer_image.migrations_django',
    'cmsplugin_filer_teaser': 'cmsplugin_filer_teaser.migrations_django',
    'cmsplugin_filer_video': 'cmsplugin_filer_video.migrations_django',
    'aldryn_gallery': 'aldryn_gallery.migrations'
}
```
which gives this error:
```
KeyError: u"Migration cmsplugin_filer_file.0001_initial dependencies reference nonexistent parent node (u'cms', u'0003_auto_20140926_2347')"
```
I noticed there are no "migrations_django" directories in either "cms" or "menus" site-packages. However changing MIGRATION_MODULES to read
```
...
    'cms': 'cms.migrations',
    'menus': 'menus.migrations',
...
```
produces:
```
django.db.utils.ProgrammingError: relation "cms_pageuser" already exists
```

Any help would be greatly appreciated.

**Comment by yakky** [comment]
Are you using Django 1.7 with both the old and new  django cms versions? You can remove ``cms`` and ``menus`` from ``MIGRATION_MODULES`` (as stated in http://django-cms.readthedocs.org/en/develop/upgrade/3.1.html#migrations-moved) 
Which is the first migration that's executed for the cms application? it should be ``0004_auto_20140924_1038.py``

**Comment by sliwowitz** [comment]
Argh! Thanks for the pointer - I actually tried to do (django-1.6, cms-3.0) -> (django-1.7, cms-3.1). So now, one step at a time, I tried to just make the move from cms-3.0 to cms-3.1 (where the first migration is 0070, and last 0076). I've hit a problem with reversion though. When I removed reversion from installed apps, I was able to migrate the rest, now I need to solve the reversion migration, which dies with this error:
```
  File "/venv/rcms/lib/python2.7/site-packages/django/template/loaders/app_directories.py", line 25, in <module>
    raise ImproperlyConfigured('ImportError %s: %s' % (app, e.args[0]))
django.core.exceptions.ImproperlyConfigured: ImportError reversion: No module named fields
```
